include/odp/api/spec/pool_ext.h

```diff
@@ -0,0 +1,318 @@
+#define ODP_API_SPEC_POOL_EXT_H_
+#include <odp/visibility_begin.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/std_types.h>
+#include <odp/api/pool.h>
+
+/** @addtogroup odp_pool
+ *  @{
+ */
+
+/**
+ * External memory pool capabilities
+ *
+ * Generic fields (not specific to a pool type) contain capabilities
+ * of the requested pool type.
+ */
+typedef struct odp_pool_ext_capability_t {
+	/** Requested pool type
+	 *
+	 *  Pool type from the odp_pool_ext_capability() call is recorded here for reference. */
+	odp_pool_type_t type;
+
+	/** Maximum number of pools
+	 *
+	 *  Maximum number of external memory pools of the requested type. */
+	uint32_t max_pools;
+
+	/** Minimum size of thread local cache */
+	uint32_t min_cache_size;
+
+	/** Maximum size of thread local cache */
+	uint32_t max_cache_size;
+
+	/** Supported statistics counters */
+	odp_pool_stats_opt_t stats;
+
+	/** Packet pool capabilities  */
+	struct {
+		/** Maximum number of packet buffers */
+		uint32_t max_num_buf;
+
+		/** Maximum packet buffer size in bytes */
+		uint32_t max_buf_size;
+
+		/** ODP header size in bytes
+		 *
+		 *  Application must reserve this many bytes from the start of a packet buffer
+		 *  for ODP implementation usage. When the value is zero, ODP implementation does
+		 *  not need header space to be reserved for it. Application will not modify this
+		 *  memory area (after buffer populate call).
+		 */
+		uint32_t odp_header_size;
+
+		/** ODP trailer size in bytes
+		 *
+		 *  Application must reserve this many bytes from the end of a packet buffer
+		 *  for ODP implementation usage. When the value is zero, ODP implementation does
+		 *  not need trailer space to be reserved for it. Application will not modify this
+		 *  memory area (after buffer populate call).
+		 */
+		uint32_t odp_trailer_size;
+
+		/** Minimum packet buffer pointer alignment in bytes
+		 *
+		 *  Packet buffer pointers populated into the pool must have at least
+		 *  this (power of two) alignment.
+		 */
+		uint32_t min_buf_align;
+
+		/** Minimum packet headroom alignment in bytes
+		 *
+		 *  Packet buffers populated into the pool must have at least this (power of two)
+		 *  alignment for the headroom.
+		 */
+		uint32_t min_head_align;
+
+		/** Packet buffer alignment flags
+		 *
+		 *  These flags specify additional alignment requirements for packet buffers.
+		 *  If not stated otherwise, min_buf_align and min_head_align alignment
+		 *  requirements apply also.
+		 */
+		struct {
+			/** Packet buffer pointers must be aligned to the buffer size.
+			  * For example, if the buffer size would be 2304 bytes (0x900),
+			  * each buffer start address must be a multiple of 0x900
+			  * (e.g. 0x12000900, 0x12001200, 0x12004800, etc). */
+			uint16_t buf_size_aligned : 1;
+		};
+
+		/** Maximum headroom parameter value
+		 *
+		 *  The packet pool headroom parameter can be configured in maximum to this value.
+		 */
+		uint32_t max_headroom;
```

Comment:
Could you please explain on what is the difference between max_headroom and max_headroom_size .

Reply:
Application requests minimum headroom size that packets have after alloc or packet input. This max_headroom capa is the maximum value for that (min) headroom parameter. Implementation may need to round up headroom size e.g. on packet input to align packet headers optimally. Max_headroom_size capa tells to application how much it needs to reserve for  headroom bytes "in the worst case", so that it is able to select large enough buffer size that fits e.g. MTU frame size in a single buffer.